interface/frontend/src/components/common/NoduleMarker.vue

```diff
@@ -38,42 +36,60 @@
         // if both new and old marker defined we've got a problem:
         // component will not detect coordinates update, so let's destroy and reinit it
         if (newMarker && oldMarker) {
-          // destroy
-          this.dynamicMarker = null
-
-          // reinit
-          var self = this
-          Vue.nextTick(function () {
-            self.dynamicMarker = newMarker
-          })
+          this.resetDynamicMarker(newMarker)
         } else {
-          this.dynamicMarker = newMarker
+          this.dynamicMarker = this.scaleMarker(newMarker)
         }
+      },
+      translation () {
+        this.resetDynamicMarker(this.marker)
       }
     },
     computed: {
-      scaledMarker () {
-        if (!this.dynamicMarker) {
+      markerOverlappedBySlice () {
+        const val = this.dynamicMarker.z !== this.sliceIndex
+        return val
+      }
+    },
+    methods: {
+      resetDynamicMarker (marker) {
+        // destroy
+        this.dynamicMarker = null
+
+        // reinit
+        var self = this
+        Vue.nextTick(function () {
+          self.dynamicMarker = self.scaleMarker(marker)
+        })
+      },
+      scaleMarker (marker) {
+        if (!marker) {
           return null
         }
 
-        var scaledInfo = {x: this.dynamicMarker.x, y: this.dynamicMarker.y, z: this.dynamicMarker.z}
+        var scaledInfo = {z: marker.z}
 
-        // converting from true DICOM pixels to display pixels
-        scaledInfo.x = scaledInfo.x * this.zoomRate - this.offsetX - this.markerImageSize / 2
-        scaledInfo.y = scaledInfo.y * this.zoomRate - this.offsetY - this.markerImageSize / 2
+        if (this.translation) {
+          // scale the position and add the offset
+          // the size of marker is subtracted to move the image center to target position
+          scaledInfo.x = marker.x * this.translation.scale + this.translation.offsetX - this.markerImageSize / 2
+          scaledInfo.y = marker.y * this.translation.scale + this.translation.offsetY - this.markerImageSize / 2
+        } else {
+          scaledInfo.x = marker.x
+          scaledInfo.y = marker.y
+        }
 
         return scaledInfo
       },
-      markerOverlappedBySlice () {
-        const val = this.scaledMarker.z !== this.sliceIndex
-        return val
-      }
-    },
-    methods: {
       onDragFinished (x, y) {
-        var scaledX = x / this.zoomRate + this.offsetX + this.markerImageSize / 2
-        var scaledY = y / this.zoomRate + this.offsetY + this.markerImageSize / 2
+        var scaledX = x
+        var scaledY = y
```

Comment:
If mutable, `let` should be preferred over `var`. However, I would suggest doing it like so:

```
const scaled = this.translation
    ? {
             x : Math.round((x + this.markerImageSize / 2 - this.translation.offsetX) / this.translation.scale),
             y : Math.round((y + this.markerImageSize / 2 - this.translation.offsetY) / this.translation.scale)
        } 
    : { x, y }
         
    EventBus.$emit('nodule-marker-moved', scaled.x, scaled.y, this.sliceIndex)

```

Reply:
Done :)